compiler/next/test/parsing/testParseForwardingDecl.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/parsing/Parser.h"
+#include "chpl/uast/Begin.h"
+#include "chpl/uast/Block.h"
+#include "chpl/uast/Comment.h"
+#include "chpl/uast/Expression.h"
+#include "chpl/uast/Identifier.h"
+#include "chpl/uast/Module.h"
+#include "chpl/queries/Context.h"
+#include "chpl/uast/ForwardingDecl.h"
+#include "chpl/uast/Record.h"
+#include "chpl/uast/VisibilityClause.h"
+
+
+// always check assertions in this test
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+#include <array>
+#include <cassert>
+
+using namespace chpl;
+using namespace uast;
+using namespace parsing;
+
+const std::string circleImpl =
+        "/* comment 1 */\n"
+        "class MyCircleImpl {\n"
+          "var radius:real;\n"
+          "proc area() {\n"
+            "return pi*radius*radius;\n"
+          "}\n"
+          "proc circumference() {\n"
+            "return 2.0*pi*radius;\n"
+          "}\n"
+        "}\n";
+
+static void test0(Parser* parser) {
+
+  const std::string myCircle =
+        "record MyCircle {\n"
+          "forwarding var impl: MyCircleImpl;\n"
+        "}\n";
+
+  const std::string circle = circleImpl + myCircle;
+
+  auto parseResult = parser->parseString("test0.chpl", circle.c_str());
+
+  assert(!parseResult.numErrors());
+  auto mod = parseResult.singleModule();
+  assert(mod);
+  assert(mod->numStmts() == 3);
+  assert(mod->stmt(0)->isComment());
+  assert(mod->stmt(1)->isClass());
+  assert(mod->stmt(2)->isRecord());
```

Comment:
It seems to be missing any checking of the uAST node for the Forwarding. You should be able to get to it by accessing the fields of the record.